slowfast/models/build.py

```diff
@@ -26,6 +27,95 @@
 """
 
 
+def get_fsdp_params(cfg) -> dict:
+    """
+    Return FSDP params dict (Assumes usage of fairscale FSDP)
+
+    Args:
+        cfg: config containing FSDP wrapping params
+
+    Return:
+        fsdp_params: FSDP params dict
+    """
+    fsdp_params = {}
+    try:
+        fsdp_params['reshard_after_forward'] = cfg.FSDP.RESHARD_AFTER_FW
+        fsdp_params["mixed_precision"] = cfg.TRAIN.MIXED_PRECISION
+    except AttributeError as e:
+        logger.exception(f"Configuration error: {e}")
+        raise e
+    if cfg.FSDP.AUTO_WRAP:
+        fsdp_params['auto_wrap_policy'] = functools.partial(
+            default_auto_wrap_policy,
+            min_num_params=int(cfg.FSDP.MIN_PARAMS_TO_WRAP)
+        )
+
+    return fsdp_params
+
+
+def fsdp_model(model: torch.nn.Module, cfg: dict, cur_device: torch.device):
+    """
+    Wraps a model with FSDP.
+
+    Args:
+        model: unwrapped torch model
+        cfg: config containing FSDP config params
+        cur_device: The device to which the model should be transfered
+
+    Return:
+        model: FSDP wrapped model
+    """
+    assert not (cfg.FSDP.AUTO_WRAP and cfg.FSDP.NESTED_WRAP), "FSDP mode: select either AUTO_WRAP or NESTED_WRAP"
+    fsdp_params = get_fsdp_params(cfg)
+
+    if cfg.FSDP.AUTO_WRAP:
+        with enable_wrap(wrapper_cls=FSDP, **fsdp_params):
+            model = auto_wrap(model)
+    elif(cfg.FSDP.NESTED_WRAP):
+        model_name = cfg.MODEL.MODEL_NAME
+        with enable_wrap(wrapper_cls=FSDP, **fsdp_params):
+            model = MODEL_REGISTRY.get(model_name)(cfg)
+
+    model = FSDP(
+            model,
+            reshard_after_forward=cfg.FSDP.RESHARD_AFTER_FW,
+            mixed_precision=cfg.TRAIN.MIXED_PRECISION,
```

Comment:
this should be reused :)

you could just use `get_fsdp_params` and remove the `auto_wrap_policy` and move it here.

Reply:
@mannatsingh Thanks, this has been resolved